### ISSUE TITLE
Add routes to communicate with PipeRTManager

### ISSUE THREAD

**Issue by DorTruzman** [ItamarWilf/PipeRT#43]
We should implement a REST API with functions in order to communicate with the GUI.
**Needed routes are the following:**

1. **Route:** GET /routines
**Return value:** JSON with "routines" key, where the value will be an array of JSONs whereas each JSON will contain the routine name and the type (if there is one).
**Possible types:** "HAS_TO_BE_FIRST", "HAS_TO_BE_LAST" or no type.
**Output example:**
```
{
    routines: [
      {
        name: "MessageFromRedis",
        type: "HAS_TO_BE_FIRST"
      },
      {
        name: "ListenToStream",
        type: "HAS_TO_BE_FIRST"
      },
      {
        name: "DoStuff"
      },
      {
        name: "DoMoreStuff"
      },
      {
        name: "MessageToRedis",
        type: "HAS_TO_BE_LAST"
      }
    ]
  }
```

2. **Route:** GET /component
**TBD**

3. **Route:** GET /routineParams/:routineName
**Return value:** JSON object where each key is a parameter and each value is a parameter type.
 **Possible types:** "String", "Integer", "QueueIn", "QueueOut", "RedisIn", "RedisOut".
**Output example (for GET /routineParams/MessageToRedis):**
```
 {
      name: "String",
      url: "String",
      max_stream_length: "Integer",
      message_queue: "QueueIn",
      redis_send_key: "RedisOut"
 }
```

3. **Route:** POST /pipeline
**Input:** Array of JSONs in the following format:
```
{
  name: COMPONENT_NAME
  queues: [ARRAY_OF_QUEUES_AS_STRINGS]
  routines: [
  {
    routine_name: ROUTINE_NAME
    PARAMETERS WILL BE HERE IN A KEY/VALUE FORMAT, SEE EXAMPLE BELOW
  }, {...}
  ]
}
```
**Return value:** 200 OK if the pipeline was saved successfully or an error message otherwise
**Input example from GUI:**
```
[
  {
    "name":"TestComponent",
    "queues":[
      "message_out"
    ],
    "routines":[
      {
        "routine_name":"MessageFromRedis",
        "name":"get_from_redis",
        "redis_read_key":"redis_read",
        "out_queue":"message_out"
      },
      {
        "routine_name":"MessageToRedis",
        "name":"send_to_redis",
        "url":"http://blabla",
        "max_stream_length":5,
        "message_queue":"message_out",
        "redis_send_key":"redis_out_key"
      }
    ]
  },
  {
    "name":"Test2Component",
    "queues":[
      "test2_out"
    ],
    "routines":[
      {
        "routine_name":"MessageFromRedis",
        "name":"get_from_test_comp",
        "redis_read_key":"redis_out_key",
        "out_queue":"test2_out"
      }
    ]
  }
]
```

4. **Route:** UPDATE /kill
**Return value:** 200 OK if the pipeline was killed successfully or an error message otherwise

Any questions can be sent here / on our WhatsApp / on VC :)

**Comment by DorTruzman** [comment]
@MayoG as I understand this is mainly a mission for you ;)

**Comment by MayoG** [comment]
Hey, couple of notes:

- `Route: GET /routines` : I made that every routine has a type and they are in enum like:
HAS_TO_BE_FIRST = 0
PROCESSING = 1
HAS_TO_BE_LAST = 2

an example of return value for me now is:
```
routines: [
      {
        name: "MessageFromRedis",
        type: 0
      },
      {
        name: "ListenToStream",
        type: 0
      },
      {
        name: "DoStuff",
        type: 1
      },
      {
        name: "DoMoreStuff",
        type: 1
      },
      {
        name: "MessageToRedis",
        type: 2
      }
    ]
```
Decide if you want me to change it or not. @DorTruzman 
(In my opinion we should return type for each routine to avoid unnecessary bugs in the future)

- Route: POST /pipeline: I changed the first key of routines will be `routine_type_name` instead of `routine_name` please change it in your code as well.

**Comment by DorTruzman** [comment]
@MayoG Good calls!
Going to implement those changes for v1.0.
If you have any more comments let me know :)

**Comment by MayoG** [comment]
Implemented in #56
